src/ManifoldProjections.jl

```diff
@@ -145,6 +160,9 @@ struct ProductManifold<:Manifold
     dims1::Tuple
     dims2::Tuple
 end
+ProductManifold(m1::Manifold, m2::Manifold, dim1::Int, dim2::Int) = ProductManifold(m1, m2, Tuple(dim1), Tuple(dim2))
+ProductManifold(m1::Manifold, m2::Manifold, dim1::Tuple, dim2::Int) = ProductManifold(m1, m2, dim1, Tuple(dim2))
+ProductManifold(m1::Manifold, m2::Manifold, dim1::Int, dim2::Tuple) = ProductManifold(m1, m2, Tuple(dim1), dim2)
```

Comment:
I'd rather not have these for the same reason as above: this API is pretty bad anyway...